src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewAddColumnDialog.cs

```diff
@@ -0,0 +1,1005 @@
+            _insertAtPosition++;
+        }
+
+        // if we persist changes directly to the designer then:
+        // 1. set the column property values
+        // 2. Add the new column
+        // 3. site the column
+
+        // 1. set the property values in the column
+        column.HeaderText = _headerTextBox!.Text;
+        column.Name = _nameTextBox!.Text;
+        column.Visible = _visibleCheckBox!.Checked;
+        column.Frozen = _frozenCheckBox!.Checked || forceColumnFrozen;
+        column.ReadOnly = _readOnlyCheckBox!.Checked;
+
+        if (_dataBoundColumnRadioButton!.Checked && _dataColumns!.SelectedIndex > -1)
+        {
+            column.DataPropertyName = ((ListBoxItem)_dataColumns.SelectedItem!).PropertyName;
+        }
+
+        if (this._persistChangesToDesigner)
+        {
+            try
+            {
+                // insert the column before siting the column.
+                // if DataGridView throws an exception then the designer does not end up w/ a column that is not in any data grid view.
+                column.DisplayIndex = -1;
+                _dataGridViewColumns.Insert(_insertAtPosition, column);
+                _insertAtPosition++;
+                // site the column
+                _liveDataGridView?.Site?.Container?.Add(column, column.Name);
+            }
+            catch (InvalidOperationException ex)
+            {
+                IUIService? uiService = _liveDataGridView?.Site?.GetService(typeof(IUIService)) as IUIService;
+                DataGridViewDesigner.ShowErrorDialog(uiService, ex, _liveDataGridView);
+                return;
+            }
+        }
+
+        // Set the UserAddedColumn property to true.
+        PropertyDescriptorCollection props = TypeDescriptor.GetProperties(column);
+        PropertyDescriptor? pd = props["UserAddedColumn"];
+        pd?.SetValue(column, true);
+
+        // pick a new Column name
+        this._nameTextBox.Text = this._headerTextBox.Text = this.AssignName();
+        this._nameTextBox.Focus();
+    }
+
+    private string AssignName()
+    {
+        int colId = 1;
+        // string columnName = (SR.DataGridView_ColumnName, colId.ToString());
+        string columnName = "Column" + colId.ToString(CultureInfo.InvariantCulture);
+
+        IDesignerHost? host = null;
+        IContainer? container = null;
+
+        host = _liveDataGridView?.Site?.GetService(iDesignerHostType) as IDesignerHost;
+        if (host is not null)
+        {
+            container = host.Container;
+        }
+
+        while (!ValidName(columnName,
+                            _dataGridViewColumns!,
+                            container,
+                            null /*nameCreationService*/,
+                            _liveDataGridView!.Columns,
+                            !_persistChangesToDesigner))
+        {
+            colId++;
+            columnName = "Column" + colId.ToString(CultureInfo.InvariantCulture);
```

Comment:
Let's use string interpolation:
```suggestion
            columnName = $"Column{colId.ToString()}";
```